frontend/src/components/Calendar/WeeklyCalendar.jsx

```diff
@@ -0,0 +1,87 @@
+import './WeeklyCalendar.scss';
+
+import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import DatePicker from '@mui/lab/DatePicker';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import { TextField } from '@mui/material';
+import startOfWeek from 'date-fns/startOfWeek';
+import { useState } from 'react';
+
+const renderDaysInWeek = (date) => {
+  const start = startOfWeek(date, { weekStartsOn: 0 });
+  let i = 0;
+  const tmpDate = start;
+  const week = [];
+  while (i !== 7) {
+    tmpDate.setDate(tmpDate.getDate() + 1);
+    week.push(
+      <div className="week-day" key={`day${i}`}>
+        <div className="day-informations">
+          <div className="day-number">{tmpDate.getDate()}</div>
+        </div>
+      </div>,
+    );
+    i += 1;
+  }
+  return week;
+};
+
+const renderDaysNames = () => {
+  const dayNames = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+  const week = [];
+  let i = 0;
+  while (i !== 7) {
+    week.push(
+      <div className="day-name" key={`day-name${i}`}>
+        {dayNames[i]}
+      </div>,
+    );
+    i += 1;
+  }
```

Comment:
how about changing while loops to return dayNames,map() - it will be a much cleaner approach.